website/project/model.py

```diff
@@ -2530,3 +2576,45 @@ def to_json(self):
                       for x in self.nodes if not x.is_deleted],
             "anonymous": self.anonymous
         }
+
+
+class Retraction(StoredObject):
+    """Retraction object for public registrations."""
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+    justification = fields.StringField(default=None, validate=MaxLengthValidator(2048))
+    initiation_date = fields.DateTimeField()
+    initiated_by = fields.ForeignField('user', backref='retracted_by')
+    approval_state = fields.DictionaryField()
+    state = fields.StringField(default='pending')
```

Comment:
Can you document the valid states? Would also be good to validate against those valid values.